src/json.rs

```diff
@@ -207,3 +252,215 @@ pub fn parse_json(data: &[u8]) -> Option<JsonValue> {
     let mut index = 0;
     rjson::parse::<JsonValue, JsonArray, JsonObject, JsonValue>(&*data_array, &mut index)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_json_type_string() {
+        let json = parse_json(r#"{"foo": "abcd"}"#.as_bytes()).unwrap();
+        let string_data = json.string("foo").ok().unwrap();
+        assert_eq!(string_data, "abcd");
+
+        let json = parse_json(r#"{"foo": 123}"#.as_bytes()).unwrap();
+        let err = json.string("foo").unwrap_err();
+        assert_eq!(err, JsonError::InvalidString);
+
+        let json = parse_json(r#"{"foo": true}"#.as_bytes()).unwrap();
+        let err = json.string("foo").unwrap_err();
+        assert_eq!(err, JsonError::InvalidString);
+
+        let json = parse_json(r#"{"foo": ["abcd"]}"#.as_bytes()).unwrap();
+        let err = json.string("foo").unwrap_err();
+        assert_eq!(err, JsonError::InvalidString);
+
+        let json = parse_json(r#"{"foo": {}}"#.as_bytes()).unwrap();
+        let err = json.string("foo").unwrap_err();
+        assert_eq!(err, JsonError::InvalidString);
+
+        let json = parse_json(r#"{"foo": null}"#.as_bytes()).unwrap();
+        let err = json.string("foo").unwrap_err();
+        assert_eq!(err, JsonError::InvalidString);
+
+        let json = JsonValue::Null;
+        let err = json.string("foo").unwrap_err();
+        assert_eq!(err, JsonError::NotJsonType);
+    }
+
+    #[test]
+    fn test_json_type_u64() {
+        let json = parse_json(r#"{"foo": 123}"#.as_bytes()).unwrap();
+        let val = json.u64("foo").ok().unwrap();
+        assert_eq!(val, 123);
+
+        let json = parse_json(r#"{"foo": 12.99}"#.as_bytes()).unwrap();
+        let val = json.u64("foo").ok().unwrap();
+        assert_eq!(val, 12);
```

Comment:
It's a little awkward to me that this is the result. I think it would be nice to file a new GH issue for this and mark it as a TODO

```
let val = json.u64("foo").ok().unwrap(); // TODO(<GH issue number here>): should fail since it is not a `u64`
```